### DISCUSSION TITLE
What happens if i change my username

### DISCUSSION THREAD

**Discussion by JlidiAmir** [community/community#47015] [Other Feature Feedback, Questions, & Ideas]
i just wandered what are the issues that can be caused by changing my username ?

**Comment by lbirkert** [comment]
Hey,

this is caused because git signs the commits with the username you previously set (Aspire9o)

Fixing this is straightforward:

1. Open cmd/terminal/powershell/whatever
2. `git config --global user.name "MedAmirJlidi"`

Take care

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @JlidiAmir ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!

**Comment by airtower-luna** [comment]
You can't have two accounts with the same email address. An email address can only be connected to one account, but you can add as many email addresses to one account as you like.

The key here is the author email address recorded in commits you make. If there is a GitHub account with that email address, GitHub will show that account as the author. If you want GitHub to show @JlidiAmir as the author, you have to set `user.email` on your computer to an email address you've added to that account. See [Setting your commit email address](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address) for how to do that. Keep in mind that changing `user.email` affects only *new* commits. If you want to change it for past commits you have to [rewrite the repository history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History), but that will very likely cause trouble if you've shared it with others already.

The `user.name` setting has *nothing* to do with your GitHub account, it's simply the author name that Git will record in commits you make. GitHub will show that name if it can't connect the author email to an account, and it's visible to anyone with a copy of the repository (e.g. with `git log`).

**Reply by airtower-luna** [reply]
I described how to get your commits associated to the right account in the second paragraph. You need to have the right author email *connected to your account* for that.

**Comment by JlidiAmir** [comment]


**Reply by lbirkert** [reply]
You should mark the message that helped you not your own answer .-.

**Reply by lbirkert** [reply]
Well, then at least try to explain what you needed to do to fix your problem, if someone ever has the same problem again.

**Reply by lbirkert** [reply]
And regarding your deleted message 
`thank you so much the problem was related to a mistyped character in my email address which lead to another account of mine linked with that email.`
It looks like the fix was just that easy.
